lighthouse.go

```diff
@@ -764,7 +762,7 @@ func (lh *LightHouse) LhUpdateWorker(ctx context.Context, f EncWriter) {
 	}
 
 	clockSource := time.NewTicker(time.Second * time.Duration(interval))
-	updateCtx, cancel := context.WithCancel(ctx)
+	updateCtx, cancel := context.WithCancel(lh.ctx)
 	lh.updateCancel = cancel
```

Comment:
This lh variable update must be synchronized. (I think that's why the test failed - the lh.updateCancel variable was read during the test's subsequent update in the main test goroutine before this line wrote the new variable in the spawned goroutine.)

I guess this could be solved by making this portion of the function synchronous, and spawning the goroutine within this function a few lines down for the actual worker loop.